provisioner/legacy_etcd.go

```diff
@@ -0,0 +1,115 @@
+package provisioner
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/kms"
+	"github.com/zalando-incubator/cluster-lifecycle-manager/api"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// populateEncryptedEtcdValues is a horrible hack that is needed to avoid doing an update of the Taupage-based CloudFormation stack every time
+// we run. Unfortunately it's pretty much unavoidable because we rely on the KMS-encrypted config items inside the launch template, and if we
+// didn't do this, every run would create new ciphertext for the same plaintext value and then trigger a rolling update. We can drop it and switch
+// to a saner scheme (same as what we do with Kubernetes) once we migrate away from Taupage, and then this whole mess can be dropped.
+func populateEncryptedEtcdValues(adapter *awsAdapter, cluster *api.Cluster, etcdKMSKeyARN string, values map[string]interface{}) error {
+	stack, err := adapter.getStackByName(etcdStackName)
+	if err != nil && !isDoesNotExistsErr(err) {
+		return err
+	}
+
+	// Try to reload the values from the existing launch template
+	if stack != nil {
+		for _, output := range stack.Outputs {
+			if aws.StringValue(output.OutputKey) == "LaunchTemplateId" {
```

Comment:
Is this the right name and not `etcd-cluster-etcd:launch-template-id` which is defined as the export name in the stack?

Reply:
The export name is stored in `output.ExportName`. Since it's not really used anywhere else, I've decided to use the key in CLM, but I don't really have a strong preference here. Should I change it to use the export name?

Reply:
No, it's fine, thanks for clarifying!